### ISSUE TITLE
Huge number of queries performed on lists due to fields

### ISSUE THREAD

**Issue by klas** [joomla/joomla-cms#14912]
### Steps to reproduce the issue
Add 10 fields and save few articles with them. Create a list with 100 items on frontend and display it. 

### Expected result
Single query for all items&fields.

### Actual result
1000 queries.

**Comment by brianteeman** [comment]
See #14558

**Comment by klas** [comment]
After #14558 there will be still 100 queries, which is not acceptable,

**Comment by mbabker** [comment]
Reopening.  The PR is an improvement but there's still room for better.

**Comment by laoneo** [comment]
As @Bakual pointe out [here](https://github.com/joomla/joomla-cms/pull/14558#issuecomment-289069161), actually it is not really possible with the current plugin events. Perhaps we can think about to make some list events in J4.

**Comment by laoneo** [comment]
I'm not sure if we want to change then tags to work like fields and use the same events there as well as it is the same problem with tags.

**Comment by csthomas** [comment]
If you want to get all fields for articles in one query then IMO you have to use some magic:)
Ex:
Instead `getFieldValue()` use `getField()` to aggregate all fields  and then you have to use magic method `__get` for `$field->value`.
This way, when you start using first value you load all fields.

Off course this is more complicated solution and not always acceptable.

**Comment by brianteeman** [comment]
Thank you for raising this issue. 

Joomla 3 is now in security only mode with no further bug fixes or new features.

As this issue doesn't relate to Joomla 4 it will now been closed. 

If we are mistaken and this does apply to Joomla 4 please open a new issue (and reference this one if you wish) with updated details for testing in Joomla 4.
cc @zero-24
